data/crac-creation/crac-creator-csa-profiles/src/main/java/com/powsybl/openrao/data/craccreation/creator/csaprofile/craccreator/remedialaction/CsaProfileRemedialActionsCreator.java

```diff
@@ -295,4 +301,141 @@ enum RemedialActionType {
         PST_RANGE_ACTION,
         NETWORK_ACTION
     }
+
+    private Set<String> createRemedialActionGroups() {
+        Set<String> standaloneRasImplicatedIntoAGroup = new HashSet<>();
+        Map<String, Set<PropertyBag>> remedialActionDependenciesByGroup = elementaryActionsHelper.getRemedialActionDependenciesByGroup();
+        elementaryActionsHelper.getRemedialActionGroupsPropertyBags().forEach(propertyBag -> {
+
+            String groupId = propertyBag.get(MRID);
+            String groupName = propertyBag.get(REMEDIAL_ACTION_NAME) == null ? groupId : propertyBag.get(REMEDIAL_ACTION_NAME);
+            try {
+                Set<PropertyBag> dependingEnabledRemedialActions = remedialActionDependenciesByGroup.get(groupId).stream().filter(raDependency -> raDependency.get(NORMAL_ENABLED).equals("true") || raDependency.get(NORMAL_ENABLED) == null).collect(Collectors.toSet());
+                if (!dependingEnabledRemedialActions.isEmpty()) {
+
+                    PropertyBag refRemedialActionDependency = dependingEnabledRemedialActions.iterator().next();
+                    String refRemedialActionDependencyKind = refRemedialActionDependency.get(RA_KIND);
+                    if (!dependingEnabledRemedialActions.stream().allMatch(raDependency -> raDependency.get(RA_KIND).equals(refRemedialActionDependencyKind))) {
```

Comment:
Create a constant called `DEPENDENCY_KIND` and use it instead of `RA_KIND` here because it can be confusing with the RemedialAction.kind attribute